chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+
+A third strategy is to use a **filter** to screen predictors before adding them to the model. For example, the analysis shown in @fig-delivery-increases computed how each item in a food order impacted the delivery time.  We could have applied a filter where we only used food item predictors whose lower confidence interval for the increase in time was greater than one. The idea behind the filter is that it is applied once prior to the model fit. 
+
+The next few sections will describe the mechanics of these techniques. However, how we utilize these algorithms is tricky. We definitely want to avoid overfitting the predictor set to the training data. 
+
+## Automatic Selection {#sec-automatic-selection}
+
+```{r}
+#| label: mars-deliveries
+#| include: false
+
+set.seed(3838)
+mars_res <- mars(prod_degree = 2, mode = "regression") %>%
+  fit(time_to_delivery ~ ., data = delivery_train)
+
+pat <- paste0("(", levels(delivery_train$day), ")", collapse = "|")
+
+mars_imp <- mars_res$fit %>%
+  evimp() %>%
+  unclass() %>%
+  as.matrix()
+
+mars_df <- mars_imp %>%
+  as_tibble(rownames = "term") %>%
+  mutate(predictor = gsub(pat, "", term))
+
+mars_active <- distinct(mars_df, predictor)
+```
+
+To more closely demonstrate how automatic feature selection works, let’s return to the Cubist model fit to the delivery data in @sec-model-development-whole-game. That model created a series of regression trees, and converted them to rules (i.e., a path through the tree), then fit regression models for every rule. Recall that there was a very large number of rules (`r format(nrow(all_rules), big.mark = ",")`) each with its own linear regression. Despite the size of the rule set, only `r length(cb_active)` of the original `r length(cb_active) + cb_unused` predictors in the data set were used in rules or model fits; `r cb_unused` were judged to be irrelevant for predicting delivery times. 
+
+::: {.important-box}
+This shows that we can often reduce the number of features for free. 
+:::
+
+Different models view the training data differently, and the list of relevant predictors will likely change from model to model. When we fit a MARS model (@sec-mars-reg) to the data, only `r nrow(mars_active)` original predictors were used. If we use automatic selection methods to understand what is important, we should fit a variety of different models and create a consensus estimate of which predictors are "active" for the data set.  
+
+## Wrapper Methods  {#sec-wrappers}
+
+Wrapper methods [@kohavi1998wrapper] use an overarching search method to optimize binary indicators, one for each potential predictor, that control which predictors are given to the model.
```

Comment:
```suggestion
Wrapper methods [@kohavi1998wrapper] use an overarching search method to optimize **binary indicators**, one for each potential predictor, that control which predictors are given to the model.
```

Very first time i read this i thought for a moment we were talking dummy variables